src/lib/rules/strict-transport-security/strict-transport-security.ts

```diff
@@ -0,0 +1,221 @@
+        };
+
+        // STS header Syntax:
+        // Strict-Transport-Security: max-age=<expire-time>
+        // Strict-Transport-Security: max-age=<expire-time>; includeSubDomains
+        // Strict-Transport-Security: max-age=<expire-time>; preload
+        // This function accomplishes the following:
+        // "max-age=31536000; includesubdomains; preload" => {"max-age":31536000,"includesubdomains":true,"preload":true}
+        const parse = (headerValue: string) => {
+            const parsedHeader = {};
+            const directives = headerValue.toLowerCase().split(';');
+            const nameValuePairRegex = /^ *([!#$%&'*+.^_`|~0-9A-Za-z-]+) *= *("(?:[~0-9])*"|[!#$%&'*+.^_`|~0-9]+) *$/;
+            // Regex for name-value pairs. E.g.: max-age=31536000
+            // Modified usage of https://github.com/jshttp/content-type/blob/64bde0d996ccb4334341662c0c7d25f7b370c4d9/index.js#L23
+            const tokenRegex = /^ *[!#$%&'*+.^_`|~0-9A-Za-z-]+$/; // Regex for single tokens. E.g.:  includesubdomains
+
+            directives.forEach((directive) => {
+                const match = tokenRegex.exec(directive) || nameValuePairRegex.exec(directive);
+
+                if (!match) {
+                    throw new Error(`'strict-transport-security' header has the wrong format: ${directive}`);
+                }
+
+                const [matchString, key, value] = match;
+                const name = key || matchString.trim();
+
+                if (parsedHeader[name]) {
+                    throw new Error(`'strict-transport-security' header contains more than one '${name}'`);
+                }
+
+                parsedHeader[name] = value || true;
+            });
+
+            return parsedHeader;
+        };
+
+        const isInvalidValue = (maxAge: string, limit: number): boolean => {
+            return maxAge && parseInt(maxAge) < limit;
+        };
+
+        const isPreloaded = async (hostname: string): Promise<boolean> => {
+            let status;
+
+            try {
+                ({ status } = await requestJSONAsync(`${statusApiEndPoint}${hostname}`));
+            } catch (err) {
+                logger.error('Error with preload verification', err);
+            }
+
+            return status === `preloaded`;
+        };
+
+        const issuesToPreload = (hostname: string): Promise<{ [key: string]: any }> => {
+            return requestJSONAsync(`${preloadableApiEndPoint}${hostname}`);
+        };
+
+        const verifyPreload = async (resource): Promise<{ [key: string]: any }> => {
+            const originalDomain = url.parse(resource).hostname;
+            const mainDomain = originalDomain.replace('www.', '');
+            const hostInTheListAlready = await isPreloaded(mainDomain) || await isPreloaded(originalDomain);
+            // Some hostnames in the list include `www.`, e.g., `www.gov.uk`.
+
+            let issues = {};
+
+            if (!hostInTheListAlready) {
```

Comment:
@alrra @molant Handling of service failure added. I wanted to add some tests with service failure mocked, but then realized that we don't have access to port number in tests, and thus the messages are never right. So I left those code commented out in case we will have access to port number in tests in the future.